kernel/src/hil/flash.rs

```diff
@@ -0,0 +1,28 @@
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum Error {
+}
+
+/// A block of writable persistent flash memory.
+pub trait Flash {
+    /// Integer type the size of a word.
+    type Word;
+
+    /// Set the client for this flash peripheral. The client will be called
+    /// when operations complete.
+    fn set_client(&self, client: &'static Client);
+
+    /// Read data
+    fn read(&self, offset: usize, buf: &mut [Self::Word]);
+
+    /// Write data
+    fn write(&self, offset: usize, buf: &mut [Self::Word]);
```

Comment:
Yes, @emosenkis is right. It needs to be `&mut` so the callback can pass it back as `&mut` to the application.